packages/react-reconciler/src/ReactFiberCompleteWork.js

```diff
@@ -1070,9 +1071,11 @@ function completeWork(
             ) {
               // We need to delete the row we just rendered.
               // Ensure we transfer the update queue to the parent.
-              // TODO: This just reuses the code from the other path but could
-              // be optimized better.
-              hasSuspendedChildrenAndNewContent(workInProgress, renderedTail);
+              let newThennables = suspended.updateQueue;
+              if (newThennables !== null) {
+                workInProgress.updateQueue = newThennables;
```

Comment:
Is this approach something we'll want to rethink later? By always attaching thenables to the SuspenseList instead of the inner boundary, perhaps.

Reply:
We could possibly do that in throwException, yea.